### ISSUE TITLE
Fix inter-pod affinity/anti-affinity design doc

### ISSUE THREAD

**Issue by Nuru** [kubernetes/community#4801]
The Pod Affinity proposal has [an inconsistency](https://github.com/kubernetes/community/blame/master/contributors/design-proposals/scheduling/podaffinity.md#L134-L136) introduced in a [cleanup](https://github.com/kubernetes/kubernetes/pull/20124) in response to a [review comment](https://github.com/kubernetes/kubernetes/pull/18265/files#r50640427). 

The proposal is linked to from the [current documentation website](https://github.com/kubernetes/website/blame/master/content/en/docs/concepts/scheduling-eviction/assign-pod-node.md#L397-L399), so I consider it part of the current documentation, and would like to see it corrected.

The inconsistency is about the interpretation of the `Namespaces` field of the `PodAffinityTerm`. The current text says:

```go
	// nil list means "this pod's namespace," empty list means "all namespaces"
	// The json tag here is not "omitempty" since we need to distinguish nil and empty.
	// See https://golang.org/pkg/encoding/json/#Marshal for more details.
	Namespaces []api.Namespace  `json:"namespaces,omitempty"`
```

The document says both that "omitempty" is omitted in order to distinguish nil and empty, which are treated differently, then includes "omitempty" in the declaration.

The current [actual code](https://github.com/kubernetes/kubernetes/blob/dd552e2059cdcc1c6da43c8c57df5f68d79b862b/pkg/apis/core/types.go#L2486-L2489) says:
```go
	// namespaces specifies which namespaces the labelSelector applies to (matches against);
	// null or empty list means "this pod's namespace"
	// +optional
	Namespaces []string
```
 The actual code looks like it has removed JSON tagging entirely, making it hard to know for sure how an empty list is encoded. 

Also, it looks like it was an [explicit design choice](https://github.com/kubernetes/kubernetes/pull/43271) to not allow a pod to specify "all namespaces", at least [for now](https://github.com/kubernetes/kubernetes/pull/43525#issuecomment-294106389), so that should be noted. 

@davidopp Are you still active and interested in fixing this?

**Comment by mrbobbytables** [comment]
Thanks for catching that @Nuru -- However it would probably be better to remove the link and/or correct it in the website documentation. =/

Design proposals are currently in an archived state and no longer being updated (PRs updating them are blocked by CI). They're place in the Kubernetes ecosystem has been superseded by [Kubernetes Enhancement Proposals](https://github.com/kubernetes/enhancements).

**Comment by cblecker** [comment]
As @mrbobbytables mentioned, design docs are the original point in time designs.. and the website documentation says that they "contain extra background information about these features".

/close
